### PR TITLE
Interactive adapters for AKRO and ADEL Staking

### PR DESCRIPTION
**Pull request by AlexanderMazaletskiy**

### CODE REVIEW

File: contracts/interactiveAdapters/Akropolis/AkroStakingInteractiveAdapter.sol [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+     * @return tokensToBeWithdrawn Array with one element - ADEL token.
+     * @dev Implementation of InteractiveAdapter function.
+     */
+    function withdraw(TokenAmount[] calldata tokenAmounts, bytes calldata)
+        external
+        payable
+        override
+        returns (address[] memory tokensToBeWithdrawn)
+    {
+        require(tokenAmounts.length == 1, "AKROIA: should be 1 tokenAmount[2]");
```
Comment (cr-igor) [inline contracts/interactiveAdapters/Akropolis/AkroStakingInteractiveAdapter.sol:104]: `tokenAmounts` should be empty and `tokensToBeWithdrawn` should be encoded in `bytes data` variable 
decode it like line 79

File: contracts/interactiveAdapters/Akropolis/AkroStakingInteractiveAdapter.sol [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        payable
+        override
+        returns (address[] memory tokensToBeWithdrawn)
+    {
+        require(tokenAmounts.length == 1, "AKROIA: should be 1 tokenAmount[1]");
+
+        address token = tokenAmounts[0].token;
+        uint256 amount = getAbsoluteAmountDeposit(tokenAmounts[0]);
+
+        ERC20(token).safeApprove(STAKING, amount, "AKROIA");
```
Comment (cr-igor) [inline contracts/interactiveAdapters/Akropolis/AkroStakingInteractiveAdapter.sol:76]: use `SafeERC20.safeApproveMax`

File: contracts/interactiveAdapters/Akropolis/AkroStakingInteractiveAdapter.sol [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+     */
+    function deposit(TokenAmount[] calldata tokenAmounts, bytes calldata data)
+        external
+        payable
+        override
+        returns (address[] memory tokensToBeWithdrawn)
+    {
+        require(tokenAmounts.length == 1, "AKROIA: should be 1 tokenAmount[1]");
+
+        address token = tokenAmounts[0].token;
```
Comment (cr-igor) [inline contracts/interactiveAdapters/Akropolis/AkroStakingInteractiveAdapter.sol:73]: add `require(token == AKRO)`

File: contracts/interactiveAdapters/Akropolis/AdelStakingInteractiveAdapter.sol [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+     */
+    function deposit(TokenAmount[] calldata tokenAmounts, bytes calldata data)
+        external
+        payable
+        override
+        returns (address[] memory tokensToBeWithdrawn)
+    {
+        require(tokenAmounts.length == 1, "ADELIA: should be 1 tokenAmount[1]");
+
+        address token = tokenAmounts[0].token;
```
Comment (cr-igor) [inline contracts/interactiveAdapters/Akropolis/AdelStakingInteractiveAdapter.sol:73]: add `require(token == ADEL)`

File: contracts/interactiveAdapters/Akropolis/AdelStakingInteractiveAdapter.sol [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+        uint256 amount = getAbsoluteAmountDeposit(tokenAmounts[0]);
+
+        ERC20(token).safeApprove(STAKING, amount, "ADELIA");
```
Comment (cr-igor) [inline contracts/interactiveAdapters/Akropolis/AdelStakingInteractiveAdapter.sol:76]: use `SafeERC20.safeApproveMax`

File: contracts/interactiveAdapters/Akropolis/AdelStakingInteractiveAdapter.sol [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+     */
+    function withdraw(TokenAmount[] calldata tokenAmounts, bytes calldata)
+        external
+        payable
+        override
+        returns (address[] memory tokensToBeWithdrawn)
+    {
+        require(tokenAmounts.length == 1, "ADELIA: should be 1 tokenAmount[2]");
+
+        address token = tokenAmounts[0].token;
```
Comment (cr-igor) [inline contracts/interactiveAdapters/Akropolis/AdelStakingInteractiveAdapter.sol:106]: decode token address from `bytes data`, use empty `tokenAmounts`

File: contracts/interactiveAdapters/Akropolis/AdelStakingInteractiveAdapter.sol [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    }
+
+    /**
+     * @notice Withdraws tokens from the Adel Staking
+     * @param tokenAmounts Array with one element - TokenAmount struct with
+     *  ADEL address, ADEL amount to be unstake
+     * @return tokensToBeWithdrawn Array with one element - ADEL token.
+     * @dev Implementation of InteractiveAdapter function.
+     */
+    function withdraw(TokenAmount[] calldata tokenAmounts, bytes calldata)
```
Comment (cr-igor) [inline contracts/interactiveAdapters/Akropolis/AdelStakingInteractiveAdapter.sol:98]: explain in docs that it always withdraws all the tokens

**Review by cr-igor — CHANGES_REQUESTED** [review]

**Comment by cr-igor** [comment]
also please remove `yarn.lock` from PR :)

File: contracts/interactiveAdapters/Akropolis/AdelStakingInteractiveAdapter.sol (hunk 5 above)
Comment (AlexanderMazaletskiy) [reply contracts/interactiveAdapters/Akropolis/AdelStakingInteractiveAdapter.sol:76]: @cr-igor cannot find safeApproveMax function in "../../shared/SafeERC20.sol"

File: contracts/interactiveAdapters/Akropolis/AkroStakingInteractiveAdapter.sol (hunk 2 above)
Comment (AlexanderMazaletskiy) [reply contracts/interactiveAdapters/Akropolis/AkroStakingInteractiveAdapter.sol:76]: @cr-igor cannot find safeApproveMax function in "../../shared/SafeERC20.sol"

**Comment by AlexanderMazaletskiy** [comment]
@cr-igor please, check

**Review by cr-igor — APPROVED** [review]

**Pull request merged**